keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryDeliveryConfigRepository.kt

```diff
@@ -15,9 +15,23 @@ import java.time.Instant.EPOCH
 class InMemoryDeliveryConfigRepository(
   private val clock: Clock = Clock.systemDefaultZone()
 ) : DeliveryConfigRepository {
+
   private val configs = mutableMapOf<String, DeliveryConfig>()
   private val lastCheckTimes = mutableMapOf<String, Instant>()
 
+  override fun deleteByApplication(application: String) {
+    configs
+      .values
+      .filter { it.application == application }
+      .map { it.application }
+      .singleOrNull()
+      ?.also {
+        configs.remove(it)
+        lastCheckTimes.remove(it)
+      }
+      ?: throw NoSuchElementException(application)
```

Comment:
Does it matter if the application doesn't exist? You're trying to make it not exist.

I'm always on the fence about whether a 'delete' call should fail if the object doesn't exist. It's informative, but annoying if it fails because it's already in the state you want. So just asking for your thoughts on this

Reply:
Yep, I don't know whether we should return 404 or not in case the application does not exist, any thoughts?

Reply:
I think it's appropriate to serve a 404 in this case. Does retrofit already convert this exception type into an actual 404 response?

Reply:
> I think it's appropriate to serve a 404 in this case. Does retrofit already convert this exception type into an actual 404 response?

Could always write a test to see

Reply:
Thanks for the feedback - will change it to return 404.

Reply:
Should this be `NoSuchApplication`?